docs/resources/dns_host.md

```diff
@@ -0,0 +1,142 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "bloxone_dns_host Resource - terraform-provider-bloxone"
+subcategory: "DNS"
+description: |-
+  Manages DNS Hosts.
+  A DNS Host object associates a DNS Config Profile with an on-prem host.
+  Note: This resource represents an existing backend object that cannot be created or deleted through API calls. Instead, it can only be updated. When using terraform apply the resource configuration is applied to the existing object, and no new object is created. Similarly terraform destroy removes the configuration associated with the object without actually deleting it from the backend.
+---
+
+# bloxone_dns_host (Resource)
+
+Manages DNS Hosts.
+
+A DNS Host object associates a DNS Config Profile with an on-prem host.
+
+Note: This resource represents an existing backend object that cannot be created or deleted through API calls. Instead, it can only be updated. When using terraform apply the resource configuration is applied to the existing object, and no new object is created. Similarly terraform destroy removes the configuration associated with the object without actually deleting it from the backend.
+
+## Example Usage
+
+```terraform
+data "bloxone_infra_hosts" "host" {
+  filters = {
+    display_name = "my-host"
+  }
+}
+
+resource "bloxone_dns_server" "dhcp_server" {
+  name = "my-dns-server"
+}
+
+resource "bloxone_dns_host" "example_dns_host" {
+  id     = data.bloxone_infra_hosts.host.results.0.legacy_id
+  server = bloxone_dns_server.dhcp_server.id
+
+  # Other Optional fields
+  tags = {
+    site = "Site A"
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `id` (String) The resource identifier.
+
+### Optional
+
+- `associated_server` (Attributes) Host associated server configuration. (see [below for nested schema](#nestedatt--associated_server))
```

Comment:
what is different in `associated_server` from `server`. I believe both are same, and one can be made readonly

Reply:
Made associated_server read only